citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -4312,7 +4312,7 @@ public void doReviseCitation (RunData data)
 		else
 		{
 			String citationId = (String) state.getAttribute(CitationHelper.CITATION_EDIT_ID);
-			int location = params.getInt("location");
+			int location = (Integer) state.getAttribute("location");
```

Comment:
We need to use the value in the state here as the params value and state value differ sometimes.